eclipse-scout-core/src/App.ts

```diff
@@ -266,47 +266,51 @@ export class App extends EventEmitter {
-      if (this._isSessionTimeoutStatus(vararg.status)) {
-        let url = requestOptions ? requestOptions.url : '';
-        this._handleBootstrapTimeoutError(vararg, url);
-        return;
+      // This may happen for REST resources or if a reverse proxy returned the response
+      url = ajaxError.requestOptions?.url || '';
+      message = ajaxError.errorDo?.message || `${ajaxError.jqXHR?.status} ${ajaxError.errorThrown}`;
+      if (ajaxError.errorDo?.correlationId) {
+        message += '. CorrelationId: ' + ajaxError.errorDo.correlationId;
       }
-    } else if (objects.isObject(vararg) && vararg.error) {
+    } else if (jsonError) {
       // Json based error
       // Json errors (normally processed by Session.js) are returned with http status 200
-      if (vararg.error.code === Session.JsonResponseError.SESSION_TIMEOUT) {
-        this._handleBootstrapTimeoutError(vararg.error, vararg.url);
-        return;
-      }
+      url = jsonError.url;
+      message = `JSON error code ${jsonError.error.code}`;
     }
+    this._handleBootstrapError(url, message);
 
     // Make sure promise will be rejected with all original arguments so that it can be eventually handled by this._fail
     // eslint-disable-next-line prefer-rest-params
     let args = objects.argumentsToArray(arguments).slice(1);
     return $.rejectedPromise(...args);
   }
 
-  protected _isSessionTimeoutStatus(httpStatus: number): boolean {
-    return httpStatus === 401;
-  }
-
-  protected _handleBootstrapTimeoutError(error: JQuery.jqXHR | JsonErrorResponse, url: string) {
-    $.log.isInfoEnabled() && $.log.info('Timeout error for resource ' + url + '. Reloading page...');
-    if (webstorage.getItemFromSessionStorage('scout:timeoutPageReload')) {
+  protected _handleBootstrapError(url: string, message: string) {
+    $.log.isInfoEnabled() && $.log.info('Error for resource ' + url + '. Reloading page...');
+    if (webstorage.getItemFromSessionStorage('scout:bootstrapErrorPageReload')) {
       // Prevent loop in case reloading did not solve the problem
-      $.log.isWarnEnabled() && $.log.warn('Prevented automatic reload, startup will likely fail', error, url);
-      webstorage.removeItemFromSessionStorage('scout:timeoutPageReload');
-      throw new Error('Resource ' + url + ' could not be loaded due to a session timeout, even after a page reload');
+      $.log.isWarnEnabled() && $.log.warn('Prevented automatic reload, startup will likely fail.');
+      webstorage.removeItemFromSessionStorage('scout:bootstrapErrorPageReload');
+      throw new Error('Resource ' + url + ' could not be loaded, even after a page reload. Reason: ' + message);
     }
-    webstorage.setItemToSessionStorage('scout:timeoutPageReload', true + '');
+    webstorage.setItemToSessionStorage('scout:bootstrapErrorPageReload', true + '');
```

Comment:
```suggestion
    webstorage.setItemToSessionStorage('scout:bootstrapErrorPageReload', 'true');
```

While we are changing the lines in this method, we could generally convert them to template strings.